api/test/integration/indicator-record/indicator-records.service.spec.ts

```diff
@@ -0,0 +1,364 @@
+        sourcingRecordId: UUIDv4(),
+        geoRegionId: UUIDv4(),
+        materialId: UUIDv4(),
+        tonnage: 10000,
+      };
+
+      jest.spyOn(materialsToH3sService, 'findOne').mockResolvedValue(undefined);
+
+      const testStatement = async (): Promise<any> => {
+        await indicatorRecordService.createIndicatorRecordsBySourcingRecords(
+          sourcingData,
+        );
+      };
+
+      //ACT/ASSERT
+      await expect(testStatement).rejects.toThrow(MissingH3DataError);
+      await expect(testStatement).rejects.toThrow(
+        `No H3 Data required for calculate Impact for Material with ID: ${sourcingData.materialId}`,
+      );
+    });
+
+    it('When creating indicators without provided coefficients and the material has H3 data, it should create the indicator records properly', async () => {
+      //ARRANGE
+      const indicatorPreconditions = await createPreconditions();
+
+      const sourcingData = {
+        sourcingRecordId: indicatorPreconditions.sourcingRecord2.id,
+        tonnage: indicatorPreconditions.sourcingRecord2.tonnage,
+        geoRegionId: indicatorPreconditions.sourcingLocation2.geoRegionId,
+        materialId: indicatorPreconditions.sourcingLocation2.materialId,
+      };
+
+      // The original function only expects for one MaterialTo3 data, the values don't matter
+      jest.spyOn(materialsToH3sService, 'findOne').mockResolvedValue({
+        materialId: UUIDv4(),
+        id: UUIDv4(),
+        h3DataId: UUIDv4(),
+        ...({} as any),
+      });
+      jest
+        .spyOn(
+          indicatorRecordRepository,
+          'getIndicatorRawDataByGeoRegionAndMaterial',
+        )
+        .mockResolvedValue({
+          production: 10,
+          harvestedArea: 100,
+          rawBiodiversity: 10,
+          rawCarbon: 5,
+          rawDeforestation: 20,
+          rawWater: 25,
+        });
+
+      //ACT
+      await indicatorRecordService.createIndicatorRecordsBySourcingRecords(
+        sourcingData,
+      );
+
+      //ASSERT
+      const allIndicators = await indicatorRecordRepository.find();
+      expect(allIndicators.length).toEqual(4);
+
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.DEFORESTATION,
+        indicatorPreconditions.deforestation,
+        sourcingData,
+        1000,
+        10,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.BIODIVERSITY_LOSS,
+        indicatorPreconditions.biodiversityLoss,
+        sourcingData,
+        500,
+        10,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.CARBON_EMISSIONS,
+        indicatorPreconditions.carbonEmissions,
+        sourcingData,
+        250,
+        10,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE,
+        indicatorPreconditions.waterRisk,
+        sourcingData,
+        12500,
+        10,
+      );
+    });
+  });
+
+  async function checkCreatedIndicatorRecord(
```

Comment:
Very nice approach, maybe we should extend this wherever possible

Reply:
It could be a nice to have to have some kind of generic function that checks one by one each property of a class, and depending on the input (if it's a complete set, or only a partial one) do expect checks as appropiate. Maybe jest already something like this. Although it might not be as trivial on more complex classes.